newapi/ooniapi/auth.py

```diff
@@ -124,19 +130,29 @@ def wrapper(*args, **kwargs):
                 return jerror("Authentication required", 401)
 
             # check for session expunge
-            # TODO: cache query
+            # TODO: cache query?
             query = """SELECT threshold
                 FROM session_expunge
                 WHERE account_id = :account_id """
             account_id = tok["account_id"]
             query_params = dict(account_id=account_id)
-            q = current_app.db_session.execute(query, query_params)
-            row = q.fetchone()
-            if row:
-                iat = datetime.utcfromtimestamp(tok["iat"])
-                threshold = row[0]
-                if iat < threshold:
-                    return jerror("Authentication token expired", 401)
+            if current_app.config["USE_CLICKHOUSE"]:
+                row = query_click_one_row(sql.text(query), query_params)
+                if row:
+                    threshold = row["threshold"]
+                    iat = datetime.utcfromtimestamp(tok["iat"])
+                    print((iat, threshold))
+                    if iat < threshold:
+                        return jerror("Authentication token expired", 401)
+
+            else:  # pragma: no cover
+                q = current_app.db_session.execute(query, query_params)
+                row = q.fetchone()
+                if row:
+                    threshold = row[0]
+                    iat = datetime.utcfromtimestamp(tok["iat"])
+                    if iat < threshold:
+                        return jerror("Authentication token expired", 401)
```

Comment:
It looks like these two if else branches have a fair amount of duplication.

I would suggest refactoring this as:

```
            if current_app.config["USE_CLICKHOUSE"]:
                row = query_click_one_row(sql.text(query), query_params)
            else:  # pragma: no cover
                q = current_app.db_session.execute(query, query_params)
                row = q.fetchone()
              if row:
                  threshold = row[0]
                  iat = datetime.utcfromtimestamp(tok["iat"])
                  if iat < threshold:
                      return jerror("Authentication token expired", 401)
```

Reply:
The duplication in done on purpose to leave the PG code untouched avoiding any risk of breaking it and remove it easily later on.